trace/http.go

```diff
@@ -2,19 +2,46 @@ package trace
 
 import "net/http"
 
-func Get(url string, parent *http.Request) (resp *http.Response, err error) {
-	c := &http.Client{}
-	req, err := http.NewRequest("GET", url, nil)
-	if err != nil {
-		return nil, err
+// DefaultCollector is a MemoryCollector. Trace defaults back to DefaultCollector if none given.
+var DefaultCollector = NewMemoryCollector()
+
+// Trace is used to add incoming Trace functionality.
+type Trace struct {
+	collector Collector
+}
+
+func (t *Trace) process(rw http.ResponseWriter, r *http.Request, f http.HandlerFunc) {
+	if t.collector == nil {
+		t.collector = DefaultCollector
 	}
-	SetIdIfMissing(req, parent)
-	return c.Do(req)
+
+	span := NewSpanIDFromRequest(r)
+	event := NewEvent(span, RequestReceived)
+	t.collector.Record(event)
+
+	r2 := SetHeaders(r, span)
+	f(rw, r2)
+
+	event = NewEvent(span, RequestCompleted)
+	t.collector.Record(event)
+}
+
+// Handler wraps an existing http.Handler with Trace functionality.
+func (t *Trace) Handler(h http.Handler) http.Handler {
+	f := h.ServeHTTP
+	return http.HandlerFunc(func(rw http.ResponseWriter, r *http.Request) {
+		t.process(rw, r, f)
+	})
 }
 
-func Instrument(fn http.HandlerFunc) http.HandlerFunc {
+// HandlerFunc wraps an existing http.HandlerFunc with Trace functionality.
+func (t *Trace) HandlerFunc(f http.HandlerFunc) http.HandlerFunc {
 	return func(rw http.ResponseWriter, r *http.Request) {
-		SetIdIfMissing(r, nil)
-		fn(rw, r)
+		t.process(rw, r, f)
 	}
 }
+
+// HandlerFuncNext middleware function can be used for Negroni integration.
+func (t *Trace) HandlerFuncNext(rw http.ResponseWriter, r *http.Request, next http.HandlerFunc) {
```

Comment:
The naming is a bit strange. If this is really for the negroni integration then name it so it is obvious. 
And I would go for the function taking a `*Trace` or `Trace` instead of a method on `Trace`.

So

```go
trace.Negroni(...)
```

is more straigtforward than

```go 
t.HandlerFuncNext(...)
```

Although implementing the `http.Handler` should be more than enough to integrate with almost every web framework/middleware lib out there. 

You could just modify the negroni hello world example:

```go
package main

import (
  "github.com/codegangsta/negroni"
  "net/http"
  "fmt"
)

func main() {
  mux := http.NewServeMux()
  mux.HandleFunc("/", func(w http.ResponseWriter, req *http.Request) {
    fmt.Fprintf(w, "Welcome to the home page!")
  })

  t := trace.NewTrace()
  n := negroni.Classic()
  n.UseHandler(trace.Handler(t, mux))
  n.Run(":3000")
}
```

This reduces the API surface which imo is always good. You could have a section in the readme or in `doc.go` that showcases how to integrate with a few popular libraries.